lib/pf/accounting.pm

```diff
@@ -656,6 +656,69 @@ sub _db_items {
     return @{$iter->all(undef) // []};
 }
 
+=head2 cleanup
+
+Perform cleanup of the accounting tables
+
+=cut
+
+sub cleanup {
+    my $timer = pf::StatsD::Timer->new( { sample_rate => 0.2 } );
+    my ( $expire_seconds, $batch, $time_limit ) = @_;
+    my $logger = get_logger();
+    $logger->debug( sub { "calling accounting_cleanup with time=$expire_seconds batch=$batch timelimit=$time_limit"; });
+
+    if ( $expire_seconds eq "0" ) {
+        $logger->debug("Not deleting because the window is 0");
+        return;
+    }
+    my $now        = pf::dal->now();
+
+    # Close old un-updated sessions
+    my %params = (
+        -set => { 
+            acctstoptime => \"NOW()",
+        },
+        -where => {
+            acctupdatetime => {
+                "<" => \[ 'DATE_SUB(?, INTERVAL ? SECOND)', $now, $expire_seconds ]
+            },
+            acctstoptime => undef,
+        },
+        -limit => $batch,
+        -no_auto_tenant_id => 1,
+    );
+    pf::dal::radacct->batch_update(\%params, $time_limit);
+
+    # Cleanup the radacct table
+    %params = (
+        -where => {
+            acctstarttime => {
+                "<" => \[ 'DATE_SUB(?, INTERVAL ? SECOND)', $now, $expire_seconds ]
+            },
+            acctstoptime => { "!=", undef },
+        },
+        -limit => $batch,
+        -no_auto_tenant_id => 1,
+    );
+    pf::dal::radacct->batch_remove(\%params, $time_limit);
+
+    # Cleanup the radacct_log table
+    %params = (
+        -where => {
+            timestamp => {
+                "<" => \[ 'DATE_SUB(?, INTERVAL ? SECOND)', $now, $expire_seconds ]
+            },
+        },
+        -limit => $batch,
+        -no_auto_tenant_id => 1,
+    );
+    pf::dal::radacct_log->batch_remove(\%params, $time_limit);
+
+
+    return;
+}
+
```

Comment:
Would this be better as three separate jobs to provide better granularity of cleanup.

Reply:
Not really because 2 of them are tightly related and radacct_log data without radacct data to go with it could cause issues I believe so its probably best to be consistent with these